src/main/java/se/sundsvall/casedata/service/scheduler/emailreader/EmailReaderWorker.java

```diff
@@ -47,35 +48,46 @@ public EmailReaderWorker(final MessageRepository repository, final ErrandReposit
 		this.notificationService = notificationService;
 	}
 
-	void getAndProcessEmails() {
+	@Transactional
+	public void getAndProcessEmails() {
 
 		try {
 			emailReaderClient.getEmail(emailReaderProperties.municipalityId(), emailReaderProperties.namespace())
-				.forEach(this::saveAndRemoteDelete);
+				.forEach(email ->
+
+				{
+					if (saveAndRemoteDelete(email)) {
+						deleteMail(email);
+					}
+				});
 		} catch (final Exception e) {
 			LOG.error("Error when fetching emails from EmailReader", e);
 		}
 	}
 
-	@Transactional
-	public void saveAndRemoteDelete(final Email email) {
+	private boolean saveAndRemoteDelete(final Email email) {
```

Comment:
I can't see any delete operations taking place in this method any longer. 
Rename method?